source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -138,10 +147,51 @@ Http::FilterHeadersStatus ProxyFilter::decodeHeaders(Http::RequestHeaderMap& hea
   NOT_REACHED_GCOVR_EXCL_LINE;
 }
 
-void ProxyFilter::onLoadDnsCacheComplete(const Common::DynamicForwardProxy::DnsHostInfoSharedPtr&) {
-  ENVOY_STREAM_LOG(debug, "load DNS cache complete, continuing", *decoder_callbacks_);
+void ProxyFilter::addHostAddressToFilterState(
+    const Network::Address::InstanceConstSharedPtr& address) {
+
+  if (!config_->saveUpstreamAddress()) {
+    return;
+  }
+
+  if (!decoder_callbacks_ || !address) {
+    ENVOY_LOG_MISC(warn, "Bad parameter - decoder callbacks or address");
+    return;
+  }
```

Comment:
delete, we don't typically add statements like this. I would pass address by const reference instead of shared pointer if you expect it to be not null.